pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -750,9 +751,11 @@ func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podStatus *ku
 			return
 		}
 		if currentPodMemoryUsage >= uint64(*podResources.Memory) {
-			klog.ErrorS(nil, "Aborting attempt to set pod memory limit less than current memory usage", "pod", pod.Name)
-			result.Fail(fmt.Errorf("aborting attempt to set pod memory limit less than current memory usage for pod %s", pod.Name))
-			return
+			if v1qos.GetPodQOS(pod) != v1.PodQOSGuaranteed {
```

Comment:
So if the pod is guaranteed, you want to force it to resize even if it means it will OOM? That doesn't seem right...

Reply:
True, but this is what the user asked for so end user is accountable for it for Guaranteed pod or ?

If we handle it the other way I.e. kubelet to continue with patching but changing what end user asked results to user not being aware of this mistake. 

If we leave it as is we end up with the QoS inconsistency , could we instead protect this returning infeasible perhaps ?

Reply:
Open to suggestions, might need some guidance how to update the status.

Reply:
Checking again think current implementation is doing what was asked for i.e. it sets the limit equal to the request. 

With regard the OOM, i don't think this will happen100% since currentPodMemoryUsage is not the actual memory use at the moment of the resize ( correct ? ),  so the OOM is not a certainity. In the case that user will patch with less memory than used then CRI will return an error or ? 

I think my reasoning considers 

> "The expectation is that entity such as VPA will set limits above use with reasonable buffer.",

as discussed [here](https://github.com/kubernetes/kubernetes/issues/124786#issuecomment-2143513768) 

 @vinaykul  what do you think?

Reply:
My fault, seems it is getting the current memory usage (The total amount of memory currently being used by the cgroup and its descendants.) . I missed those updates

https://github.com/kubernetes/kubernetes/blob/feb3f92bc42f006874f68c476c2d7a6f2ac5ab16/pkg/kubelet/cm/cgroup_v2_manager_linux.go#L80

Putting back to WIP, apologies for the noise.

Reply:
@tallclair PTAL , i think this should fix it.

Reply:
>Checking again think this PR is solving the issue, and in reality is doing what was asked for i.e. it sets the limit equal to the request to overcome the QoS inconsistency reported in the issue.

Can you please help me understand where the QoS inconsistency comes from? The QoS of a pod does not change once created (immutability is due to how this is setup at the kernel cgroupfs level at pod creation time) and it is recorded in PodStatus.QOSClass.

If mem.req and mem.lim do not match in the status because limit cannot be set due to usage, that is the right behavior imo. The impact of such a configuration is that the pod becomes a more attractive candidate for eviction. What matters is for Guaranteed QoS class pod, API validation should never allow requests to deviate from req == lim for cpu & mem on all containers.

Reply:
BTW, I find the following quite useful to see everything in one screen:
```bash
kg po gpod -ojson | jq '{specResources: .spec.containers[0].resources,  allocatedResources: .status.containerStatuses[0].allocatedResources, statusResources: .status.containerStatuses[0].resources, resizeStatus: .status.resize}'"
```

Reply:
> If mem.req and mem.lim do not match in the status because limit cannot be set due to usage, that is the right behavior imo. 

Looks good to me, since resize status is InProgress and QoS class is not changed, we can close the issue then without any PR. WDYT @tallclair